### ISSUE TITLE
test runner runs all tests above the one which is marked as .only

### ISSUE THREAD

**Issue by Zombobot1** [oven-sh/bun#10850]
### What version of Bun is running?

1.1.7+b0b7db5c0

### What platform is your computer?

Darwin 23.4.0 arm64 arm

### What steps can reproduce the bug?

```ts
// test.ts
import { describe, it, expect } from 'bun:test'

describe('describe', () => { // without describe it doesn't work as expected either
  // same for test instead of it
  it('test 1', () => {
    expect(1).toBe(1) // also runs
  })

  it.only('test 2', () => {
    expect(2).toBe(2)
  })
})
```

`bun test ./test.ts`

```
✓ describe > test 1 [0.03ms]
✓ describe > test 2 [0.01ms]
```

### What is the expected behavior?

test runner doesn't run tests which are not marked as only (to match jest and vitest behavior)

### What do you see instead?

test runner runs all tests above the one which is marked as only (but tests below do not run 🫤)

**Comment by ilbertt** [comment]
The same bug is happening to me.

Bun revision: `1.1.12+43f0913c3`
Computer platform: `Darwin 23.5.0 arm64 arm`

**Comment by ilbertt** [comment]
@Zombobot1 I think the solution is to use the `--only` flag: https://github.com/oven-sh/bun/issues/5893#issuecomment-1731810341

**Comment by EvHaus** [comment]
Looks like this is fixed for me in bun v1.1.21

**Comment by Jarred-Sumner** [comment]
This will be fixed in Bun v1.1.26
